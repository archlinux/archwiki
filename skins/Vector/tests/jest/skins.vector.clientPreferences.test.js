const clientPreferences = require( '../../resources/skins.vector.clientPreferences/clientPreferences.js' );

let cp;
describe( 'clientPreferences', () => {
	beforeEach( () => {
		document.body.innerHTML = '';
		cp = document.createElement( 'div' );
		cp.id = 'cp';
		document.body.appendChild( cp );
		mw.requestIdleCallback = ( callback ) => callback();
		mw.user.clientPrefs = {
			get: jest.fn( () => '1' )
		};
		const portlet = document.createElement( 'div' );
		mw.util.addPortlet = jest.fn( () => portlet );
		mw.util.addPortletLink = jest.fn( () => {
			const li = document.createElement( 'li' );
			const a = document.createElement( 'a' );
			li.appendChild( a );
			portlet.appendChild( li );
			return li;
		} );
		mw.message = jest.fn( ( key ) => ( {
			text: () => `msg:${ key }`,
			exists: () => true
		} ) );
	} );

	test( 'render empty', () => {
		return clientPreferences.render( '#cp', {} ).then( () => {
			expect( cp.innerHTML ).toMatchSnapshot();
		} );
	} );

	test( 'render font size', () => {
		document.documentElement.setAttribute( 'class', 'vector-feature-custom-font-size-clientpref-2' );
		return clientPreferences.render( '#cp', {
			'vector-feature-custom-font-size': {
				options: [ '0', '1', '2' ],
				preferenceKey: 'vector-font-size'
			}
		} ).then( () => {
			expect( cp.innerHTML ).toMatchSnapshot();
		} );
	} );

	test( 'render toggle', () => {
		document.documentElement.setAttribute( 'class', 'expandAll-clientpref-1' );
		return clientPreferences.render( '#cp', {
			expandAll: {
				options: [ '0', '1' ],
				preferenceKey: 'expandAll',
				type: 'switch'
			}
		} ).then( () => {
			expect( cp.innerHTML ).toMatchSnapshot();
		} );
	} );
} );
