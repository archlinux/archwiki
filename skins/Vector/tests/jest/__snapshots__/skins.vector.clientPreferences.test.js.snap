// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clientPreferences render empty 1`] = `""`;

exports[`clientPreferences render font size 1`] = `"<div><li><div class=\\"\\"><form><div class=\\"cdx-radio\\"><input name=\\"skin-client-pref-vector-feature-custom-font-size-group\\" id=\\"skin-client-pref-vector-feature-custom-font-size-value-0\\" type=\\"radio\\" value=\\"0\\" data-event-name=\\"skin-client-pref-vector-feature-custom-font-size-value-0\\" class=\\"cdx-radio__input\\"><span class=\\"cdx-radio__icon\\"></span><label for=\\"skin-client-pref-vector-feature-custom-font-size-value-0\\" class=\\"cdx-radio__label\\">vector-feature-custom-font-size-0-label</label></div><div class=\\"cdx-radio\\"><input name=\\"skin-client-pref-vector-feature-custom-font-size-group\\" id=\\"skin-client-pref-vector-feature-custom-font-size-value-1\\" type=\\"radio\\" value=\\"1\\" data-event-name=\\"skin-client-pref-vector-feature-custom-font-size-value-1\\" class=\\"cdx-radio__input\\"><span class=\\"cdx-radio__icon\\"></span><label for=\\"skin-client-pref-vector-feature-custom-font-size-value-1\\" class=\\"cdx-radio__label\\">vector-feature-custom-font-size-1-label</label></div><div class=\\"cdx-radio\\"><input name=\\"skin-client-pref-vector-feature-custom-font-size-group\\" id=\\"skin-client-pref-vector-feature-custom-font-size-value-2\\" type=\\"radio\\" value=\\"2\\" data-event-name=\\"skin-client-pref-vector-feature-custom-font-size-value-2\\" class=\\"cdx-radio__input\\"><span class=\\"cdx-radio__icon\\"></span><label for=\\"skin-client-pref-vector-feature-custom-font-size-value-2\\" class=\\"cdx-radio__label\\">vector-feature-custom-font-size-2-label</label></div></form></div></li></div>"`;

exports[`clientPreferences render toggle 1`] = `"<div><li><div class=\\"\\"><form><span class=\\"cdx-toggle-switch\\"><input name=\\"skin-client-pref-expandAll-group\\" id=\\"skin-client-pref-expandAll-value-1\\" type=\\"checkbox\\" data-event-name=\\"skin-client-pref-expandAll-value-1\\" class=\\"cdx-toggle-switch__input\\"><span class=\\"cdx-toggle-switch__switch\\"><span class=\\"cdx-toggle-switch__switch__grip\\"></span></span><label class=\\"cdx-toggle-switch__label\\">msg:expandAll-name</label></span></form></div></li></div>"`;
